pkg/scheduler/ugm/manager.go

```diff
@@ -419,30 +348,40 @@ func (m *Manager) UpdateConfig(config configs.QueueConfig, queuePath string) err
 	m.Lock()
 	defer m.Unlock()
 
-	// clear the local limit config maps before processing the limit config
-	m.userLimitsConfig = make(map[string]map[string]map[string]interface{})
-	m.groupLimitsConfig = make(map[string]map[string]map[string]interface{})
+	m.userWildCardLimitsConfig = make(map[string]*LimitConfig)
+	m.groupWildCardLimitsConfig = make(map[string]*LimitConfig)
 	return m.internalProcessConfig(config, queuePath)
 }
 
 func (m *Manager) internalProcessConfig(cur configs.QueueConfig, queuePath string) error {
 	// Holds user and group for which limits have been configured with specific queue path
-	userGroupLimits := make(map[string]bool)
+	userLimits := make(map[string]bool)
+	groupLimits := make(map[string]bool)
 	// Traverse limits of specific queue path
 	for _, limit := range cur.Limits {
-		tempLimitsMap := make(map[string]interface{})
-		tempLimitsMap[maxresources] = limit.MaxResources
-		tempLimitsMap[maxapplications] = limit.MaxApplications
+		var maxResource *resources.Resource
+		var err error
+		if maxResource, err = resources.NewResourceFromConf(limit.MaxResources); err != nil {
+			log.Log(log.SchedUGM).Debug("Problem in using the wild card limit max resources settings.",
```

Comment:
Assume all resources can be parsed: the config must have passed the  validation before we get here